node/chunk_pool/src/mem_pool/chunk_pool_inner.rs

```diff
@@ -204,33 +204,28 @@ impl MemoryChunkPool {
 
     /// Updates the cached file info when log entry retrieved from blockchain.
     pub async fn update_file_info(&self, tx: &Transaction) -> Result<bool> {
-        let mut inner = self.inner.lock().await;
-
-        // Do nothing if file not uploaded yet.
-        let file = match inner.segment_cache.get_file_mut(&tx.data_merkle_root) {
-            Some(f) => f,
-            None => return Ok(false),
-        };
-
-        // Update the file info with transaction.
-        file.update_with_tx(tx);
-
-        // File partially uploaded and it's up to user thread
-        // to write chunks into store and finalize transaction.
-        if file.cached_chunk_num < file.total_chunks {
-            return Ok(true);
-        }
-
-        // Otherwise, notify to write all memory cached chunks and finalize transaction.
-        let file_id = FileID {
-            root: tx.data_merkle_root,
-            tx_id: tx.id(),
-        };
-        if let Err(e) = self.sender.send(file_id) {
-            // Channel receiver will not be dropped until program exit.
-            bail!("channel send error: {}", e);
+        let maybe_file = self
+            .inner
+            .lock()
+            .await
+            .segment_cache
+            .remove_file(&tx.data_merkle_root);
+        if let Some(mut file) = maybe_file {
+            file.update_with_tx(tx);
+            for (seg_index, seg) in file.segments.into_iter() {
+                self.write_chunks(
```

Comment:
For DA scenario, it is ok to store partial segments into db. However, the perf may be low and block the log entry sync thread.

On the other hand, small file will not be finalized if user uploaded the entire file before log entry retrieved from blockchain node.

Reply:
monitor_log_entry is run in another thread and won't block log entry sync.

Small files will finalize, there is file finalization check at the end of write_chunks().